src/main/java/hudson/plugins/audit_trail/ElasticSearchAuditLogger.java

```diff
@@ -0,0 +1,460 @@
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.apache.commons.lang.time.FastDateFormat;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.apache.http.ssl.TrustStrategy;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardCertificateCredentials;
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
+import com.google.common.collect.Range;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+
+/**
+ * AuditLogger implementation to send audit logs to an Elastic Search server.
+ * Some code take from the Jenkins logstash plugin: https://github.com/jenkinsci/logstash-plugin
+ *
+ * Default values are set in <code>/src/main/resources/hudson/plugins/audit_trail/ElasticSearchAuditLogger/config.jelly</code>
+ *
+ * @author <a href="mailto:alexander.russell@sap.com">Alex Russell</a>
+ */
+public class ElasticSearchAuditLogger extends AuditLogger {
+
+    private String url;
+    private String usernamePasswordCredentialsId;
+    private String clientCertificateCredentialsId;
+    private boolean skipCertificateValidation = false;
+
+    transient ElasticSearchSender elasticSearchSender;
+
+    protected static final Logger LOGGER = Logger.getLogger(ElasticSearchAuditLogger.class.getName());
+    private static final FastDateFormat DATE_FORMATTER = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ssZ");
+
+    @DataBoundConstructor
+    public ElasticSearchAuditLogger(String esServerUrl, boolean skipCertificateValidation) {
```

Comment:
This will break JCASC, could you please rename esServerlUrl to url:

```suggestion
    public ElasticSearchAuditLogger(String url, boolean skipCertificateValidation) {
```